wordpress-mobile/WordPress-iOS
# [Notifications P1] Comment Moderation View

Refs #https://github.com/Automattic/wordpress-mobile/issues/33

## Description
This PR adds a customisable moderation view for all moderation states. This component will be used in the following Comment Moderation Tickets [#https://github.com/Automattic/wordpress-mobile/issues/33, #https://github.com/Automattic/wordpress-mobile/issues/34, #https://github.com/Automattic/wordpress-mobile/issues/35, #https://github.com/Automattic/wordpress-mobile/issues/36]

## Testing Steps
Actions are not yet implemented other than causing the state & UI to change. ViewModel is in place to allow further implementation. UI is not attached to the flow so testing is possible via Previews.

Go to `CommentModerationView.swift` file and enable canvas (comment out `didFinishLaunchingWithOptions` and `willFinishLaunchingWithOptions` functions in `WordPressAppDelegate` and select `WordPress` as build target instead of `Jetpack`). Scrolling horizontally will reveal all possible states. Each one is interactive. Verify if the designs match to yWt5gg3nWORhu079Qfv3mS-fi-1118_2402 (the button colors are updated to sync with Design System Component).

## Screenshot Reference
The tertiary button is blue instead of Jetpack green due to running `WordPress` build for Previews. The dyanmic Design System color is used.

https://github.com/wordpress-mobile/WordPress-iOS/assets/15968946/304b5757-55db-4ac3-9beb-ab59c1130b5a


## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:


Testing checklist:
